src/bodies.cpp

```diff
@@ -579,76 +579,72 @@ void bodies::Box::computeBoundingCylinder(BoundingCylinder &cylinder) const
 
 bool bodies::Box::intersectsRay(const Eigen::Vector3d& origin, const Eigen::Vector3d& dir, EigenSTL::vector_Vector3d *intersections, unsigned int count) const
 {
-  if (detail::distanceSQR(center_, origin, dir) > radius2_) return false;
-
-  double t_near = -std::numeric_limits<double>::infinity();
-  double t_far  = std::numeric_limits<double>::infinity();
 
-  for (int i = 0; i < 3; i++)
+  //Brian Smits. Efficient bounding box intersection. Ray tracing news 15(1), 2002
+  float tmin, tmax, tymin, tymax, tzmin, tzmax;
+  float divx, divy, divz;
+  divx = 1 / dir.x();
```

Comment:
Check that dir.x() is not zero

Reply:
The division by possible zero is needed here